### ISSUE TITLE
Module 5.6.0 roadmap items

### ISSUE THREAD

**Issue by arnaudlh** [aztfmod/terraform-azurerm-caf#1125]
Module version 5.6.0 will implement the following features. 

## 5.6.0
Move to azurerm 2.99
Minor refactoring

## 5.6.x
Move to microsoft graph with azuread provider 2.x - estimated time: may 2022
Move to azurerm 3.0 - estimated time: may 2022

**Comment by langecode** [comment]
Stumbled into a problem trying to do "credential federation" with Azure AD and discovered the azuread provider `1.4.0` is over 2 years old! Would be really nice to have that upgraded.

**Comment by nusrath432** [comment]
Related: https://github.com/Azure/caf-terraform-landingzones/issues/425

**Comment by LaurentLesle** [comment]
You can test the pre-release with the landingzone tag int-5.7.0.
So far the upgrade tests are working (with some modifications required in the configuration file for azure ad applications /sp and groups)
The goal is to get a module release next week

**Comment by MarcelHeek** [comment]
@LaurentLesle We are consuming the CAF supermodule using a local fork which we regularly sync with the public module (main branch). In our local fork we use an "own" branch where we maintain small changes for non AzureRM 3.x related things that are not supported in the module (yet). So just consuming the pre-release branch is not as easy as it seems.

You mention the goal is to release a module version bump next week. Will that be with AzureRM 3.x (a 5.7.x release) or an updated release for AzureRM 2.99 (a 5.6.10) release? If it is the first, my co-workers and I would be very happy.

**Comment by LaurentLesle** [comment]
5.7.0 is using latest azurerm (3.56.0) and azuread (2.39.0)
There is an azure branch pointing to the pre-release (int-5.7.0)

**Comment by MarcelHeek** [comment]
@LaurentLesle @arnaudlh As I expected the new release is 5.6.10, not 5.7.0 yet.  Is there a estimation or targetdate for the 5.7.0 version release you guys are aiming for?  I can understand that answering this might be difficult, but it is crucial for us to know what to expect so we can adjust our plans accordingly.

**Comment by mgibson85** [comment]
Hi guys, any further news on the 5.7.0 release? Like many others, we are waiting on azuread provider 2.x with graph.

Thanks for all the hard work :)
